src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -186,20 +187,34 @@ private OkObjectResult FormatFindResult(JsonDocument jsonDoc, FindRequestContext
         {
             JsonElement jsonElement = jsonDoc.RootElement.Clone();
 
+            // When there are no rows returned from the database, the jsonElement will be an empty array.
+            // In that case, the response is returned as is.
+            if (jsonElement.ValueKind is JsonValueKind.Array && jsonElement.GetArrayLength() == 0)
+            {
+                return OkResponse(jsonElement);
+            }
+
+            IEnumerable<string> extraFieldsInResponse = (jsonElement.ValueKind is not JsonValueKind.Array)
+                                                  ? DetermineExtraFieldsInResponse(jsonElement, context)
+                                                  : DetermineExtraFieldsInResponse(jsonElement.EnumerateArray().Last(), context);
+
             // If the results are not a collection or if the query does not have a next page
-            // no nextLink is needed, return JsonDocument as is
+            // no nextLink is needed. So, the response is returned after removing the extra fields.
             if (jsonElement.ValueKind is not JsonValueKind.Array || !SqlPaginationUtil.HasNext(jsonElement, context.First))
             {
-                // Clones the root element to a new JsonElement that can be
-                // safely stored beyond the lifetime of the original JsonDocument.
-                return OkResponse(jsonElement);
+                return jsonElement.ValueKind is JsonValueKind.Array ? OkResponse(JsonSerializer.SerializeToElement(RemoveExtraFieldsInResponseWithMultipleItems(jsonElement.EnumerateArray().ToList(), extraFieldsInResponse)))
+                                                                    : OkResponse(RemoveExtraFieldsInResponseWithSingleItem(jsonElement, extraFieldsInResponse));
             }
 
+            IEnumerable<JsonElement> rootEnumerated = jsonElement.EnumerateArray().ToList();
```

Comment:
Why do you need ToList() if you are using IEnumerable<T>? The original code did not use ToList()

Reply:
Removed it